build/apple_release/scripts/download-signed-mac-OS-binaries.sh

```diff
@@ -0,0 +1,75 @@
+#!/bin/bash
+
+cliExecutableName=$1
+releaseVersion=$2
+goarch=$3
+GITHUB_ACCESS_TOKEN=$4
+
+# This script downloads signed macOS binaries for a specific version and architecture.
+
+# Function to retrieve the specific artifact URL with retries
+get_specific_artifact_url_with_retries() {
+    local max_retries=4
+    local cooldown=15 # Cooldown in seconds between retries
+    local retry_count=0
+
+    while [ $retry_count -lt $max_retries ]; do
```

Comment:
@yahavi,
In here we can't use curl --retry, because the curl execution will be okay as it queries all the artifacts from the repo.
The retry is when the query doesn't return the specific version and release version executable, which we try to filter from the response.